src/MongoDB.Bson/PreserveAttribute.cs

```diff
@@ -0,0 +1,39 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.ComponentModel;
+
+namespace MongoDB.Bson
+{
+    /// <summary>
+    /// Prevents the Xamarin managed linker from linking the target.
```

Comment:
It doesn't prevent the linker from linking. It forces the linker to link the target regardless of whether it is statically referenced or not.

Reply:
You're right - this should have said `removing` instead of `linking`. That being said, it's probably not super important as the attribute is not part of the public API of the driver and users are not expected to use it.